### PR TITLE
migrations(plugins): move Slack and PagerDuty plugins from test_only_plugins to new_plugins

### PR DESCRIPTION
**Pull request by scefali**
Must be merged with https://github.com/getsentry/sentry-plugins/pull/542

Also makes plugin tests required

### CODE REVIEW

File: src/sentry/testutils/cases.py [hunk 1]
```diff
@@ -699,20 +699,6 @@ def assertPluginInstalled(self, name, plugin):
...
-                if ep_path == path:
-                    return
-                self.fail(
-                    "Found plugin in entry_points, but wrong class. Got %r, expected %r"
-                    % (ep_path, path)
-                )
-        self.fail("Missing plugin from entry_points: %r" % (name,))
-
     # TODO (Steve): remove function
     def assertNewPluginInstalled(self, name, plugin):
```
Comment (joshuarli) [inline src/sentry/testutils/cases.py]: Is this gonna be removed in a follow-up pr after this is merged?

Comment (scefali) [reply src/sentry/testutils/cases.py]: @joshuarli Yes. In the follow-up PR I will move everything from `new_sentry_plugins` into `sentry_plugins` and clean up other items.

**Review by wedamija — APPROVED** [review]

File: src/sentry/testutils/cases.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry/testutils/cases.py]: 🎉

**Comment by scefali** [comment]
Closing until ready to merge

**Pull request merged**